### ISSUE TITLE
scuff-transmission for 1d gratings

### ISSUE THREAD

**Issue by gevero** [HomerReid/scuff-em#38]
Hi Homer

I was wondering if is it possible to use scuff-em to model 1d periodic gratings. Of course something of this kind is possible using a properly formatted unit cell and the full calculation, nevertheless a purely 1d solution would be much faster, I guess. I saw that something along these lines is possible given that purely 2d casimir calculations are possible! Thanks a lot

Giovanni

**Comment by HomerReid** [comment]
Yes, it is definitely possible to model 1D periodic gratings. If you can describe the geometry in which you are interested and the results you would like to extract for this geometry, I can help you set up the calculation.

**Comment by gevero** [comment]
Hi Homer

Thanks for the quick answer. Below you can find two sample structures that I would like to simulate. Regarding the physical setup and the results I would like to obtain they are as follows:

* **Incident Field**: Plane wave with arbitrary polar and azimuthal angles. I guess that the arbitrary azimuth might be a tricky part, because in this case the problem is only pseudo 1d. The plane wave may come both from the top or the bottom of the structure.
* **Calculated quantities**: I need for sure Reflectance and Transmittance, or event better the reflection and the transmission matrix, which implies that I should be able to extract the polarization state of the reflected and transmitted fields. This is also tricky because it would be interesting to consider 2 separate cases, i.e. the first where all the diffraction orders are included, and the second where only the first diffraction order is retained. 

Thanks a lot for your time and help

Best

Giovanni

![slide1](https://cloud.githubusercontent.com/assets/7213490/6992122/bd9450e0-dabb-11e4-9c91-a3f756d9e83d.PNG)
![slide2](https://cloud.githubusercontent.com/assets/7213490/6992123/bd9b00f2-dabb-11e4-89f7-f1689463af01.PNG)
